### PR TITLE
mgmt, sql upgrade api-version to 2021-11

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

Manual interface change: 8bfbfdbdd73721e3612928f2010760d572b0f00e
Manual implementation change: 7837fad40575ca59857b7aa6d08952956cd5b64b
Manual code delete: f8dedb453225b31fb47cb159fd4fb3f4e7b4b559

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-sql/src/test/java/com/azure/resourcemanager/sql/SqlServerOperationsTests.java [hunk 1]
```diff
@@ -986,17 +987,22 @@ public void canCRUDSqlDatabase() throws Exception {
         String accountKey = storageAccount.getKeys().get(0).value();
         String blobEntrypoint = storageAccount.endPoints().primary().blob();
 
-        sqlDatabase.defineThreatDetectionPolicy(SecurityAlertPolicyName.DEFAULT)
+        List<String> disabledAlerts = Collections.singletonList("Sql_Injection");
+
+        sqlDatabase.defineThreatDetectionPolicy(SecurityAlertPolicyName.fromString("myPolicy"))
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-sql/src/test/java/com/azure/resourcemanager/sql/SqlServerOperationsTests.java:992]: Policy name seems not taking effect. Policies will always have name "Default".

**Comment by weidongxu-microsoft** [comment]
@srnagar @JonathanGiles for guidance on breaking changes.

This is actually upgrade SQL mgmt-plane from 2014 to 2021. Hence there is some API (mostly monitoring) that get removed from backend, which we are not able to avoid even with handwritten code.

Currently we still plan to continue it at 2.22.0 (no major version bump).

Alternative is to bump azure-resourcemanager-sql to 3.0.0 (major version bump), but keep the uber package azure-resourcemanager still at 2.22.0

**Comment by JonathanGiles** [comment]
@weidongxu-microsoft Do we know how other languages are dealing with this issue?

**Comment by weidongxu-microsoft** [comment]
> @weidongxu-microsoft Do we know how other languages are dealing with this issue?

We do not know how .NET is going to handle this.

Other languages (python, JS, Go) will bump up major version.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**